DESIGN.md

```diff
@@ -0,0 +1,202 @@
+🚨 DRAFT OUTLINE 🚨
+
+This document attempts to describe the general design decisions `pgx` has made as it relates to providing a Rust-y API
+around Postgres internals.
+
+## A Note from @eeeebbbbrrrr
+
+pgx was developed out of frustrations with developing Postgres extensions in C and SQL, specifically @ZomboDB. ZomboDB 
+quickly evolved into a fairly large extension.  Roughly 25k lines of C, 8k lines of hand-written SQL, and all sorts of 
+bash machinery to accommodate testing, releases, and general project maintenance.  In C, it is just too much for one 
+person to manage.  I needed a hand and the Rust compiler seemed like a good coworker.
+
+Today, ZomboDB is still about 25k lines -- but of Rust!, about 1k lines of hand-written SQL, and much less bash
+machinery.  And the Rust compiler really is a good coworker.
+
+## Prior Art
+
+- https://github.com/jeff-davis/postgres-extension.rs
+- https://github.com/bluejekyll/pg-extend-rs
+- https://github.com/wasmerio/wasmer-postgres
+
+None of these appear active today.
+
+## Goals
+
+pgx' primary goal is to make Rust extension development as natural to Rust programmers as possible.  This generally 
+means doing the best we can to avoid violating POLA as it relates to error handling, memory management, and type 
```

Comment:
The POLA acronym should probably be expanded, it's not immediately clear what it stands for ([principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)?)

Reply:
indeed.  Will do.  ;)